src/main/ruby/core/truffle/cext.rb

```diff
@@ -1151,7 +1151,7 @@ def rb_intern3(string, enc)
   end
 
   def rb_str_append(str, to_append)
-    str << StringValue(to_append)
+    str.append(to_append)
```

Comment:
What's the difference between append and `<<`?
I'd rather use standard methods if there is no difference.

Reply:
Mostly how they deal with `Fixnum` values.

```
> "abc".append('d'.ord)
TypeError: Truffle doesn't have a case for the org.truffleruby.core.string.StringNodesFactory$StringAppendPrimitiveNodeFactory$StringAppendPrimitiveNodeGen node with values of type  String(com.oracle.truffle.object.basic.DynamicObjectBasic) java.lang.Integer=100

> "abc".concat('d'.ord)
=> "abcd"
```

I can appreciate wanting to use standard methods, but I'd really rather match MRI's semantics than spend hours tracking down an issue due to some subtle change.

Reply:
FWIW, this is a non-standard method instead of a primitive because rubysl expects the method to exist. But we could just modify that in place now.

Reply:
I see, it might be clearer to explicitly check for `to_append` to be a String here.
Depending on undocumented, Rubinius-internal behavior sounds brittle and I'd like to get rid of them as it is a user-visible difference (And in 2.5, we might have String#append as alias of #<<, it's already the case for Array on 2.5).

Reply:
There's another issue here where `rb_str_append` should rap the other argument in `StringValue`.

This isn't Rubinius behavior, though. This is MRI behavior that Rubinius carried over. `rb_str_append` and `rb_str_concat` have these different semantics in MRI.

Reply:
Could we move the method to StringOperations and rename it to `rb_str_append`? Then it would be very clear. Or make it a primitive, same for me.

Reply:
Yeah. I'll do that. Like I said, it had to be on `String` because `StringIO` from ruby-sl expects to just call it. But now that we're modifying ruby-sl files in place I'll just clean it up and make it a primitive. We need it to be fast, as `String#<<` delegates to it in the common case.